File: Data Structures & Algorithms/median-of-two-sorted-arrays/submission-0.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> sum;
        int i=0, j=0;
        while(i < nums1.size() && j < nums2.size()){
            if(nums1[i] > nums2[j]){
                sum.push_back(nums2[j]);
                j++;
            }else{
                sum.push_back(nums1[i]);
                i++;
            }
        }

        while(i < nums1.size()){
            sum.push_back(nums1[i]);
            i++;
        }

        while(j < nums2.size()){
            sum.push_back(nums2[j]);
            j++;
        }

        double res;
        int len = sum.size();

        if(len % 2 == 0){
            res = ((double)sum[len / 2] + (double)sum[len / 2 - 1]) / 2;
        }else{
            res = sum[len / 2];
        }

        return res;
    }
};

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res = []
        cur = []
        def dfs(nums2):
            if len(nums2) == 1:
                cur.append(nums2[0])
                res.append(cur.copy())
                cur.pop()
                return
                
            for i in range(len(nums2)):
                cur.append(nums2[i])
                print(nums2[:i] + nums2[i+1:])
                dfs(nums2[:i] + nums2[i+1:])
                cur.pop()
            
        dfs(nums)
        return res